example breaks when `path=` is replaced with `version=`

### Summary

When attempting to reproduce the code in the `/examples/browser-webrtc` directory in a new cargo folder, the example breaks when: 

```
libp2p = { path = "../../libp2p", ...
```

is replaced with:

```
libp2p = { version = "0.53.1", ...
```

### Expected behavior

I would expect to be able to drop in the same `version =` as the lib at `path =` and compile without issue.

### Actual behavior

Example doesn't compile when `path=` replaced with `version=`

### Relevant log output

```shell
error[E0599]: the method `map` exists for struct `Transport`, but its trait bounds were not satisfied
  --> examples/browser-webrtc/src/main.rs:36:14
   |
32 |               Ok(webrtc::tokio::Transport::new(
   |  ________________-
33 | |                 id_keys.clone(),
34 | |                 webrtc::tokio::Certificate::generate(&mut thread_rng())?,
35 | |             )
36 | |             .map(|(peer_id, conn), _| (peer_id, StreamMuxerBox::new(conn))))
   | |             -^^^ method cannot be called on `Transport` due to unsatisfied trait bounds
   | |_____________|
   | 
   |
  ::: /rust-libp2p/transports/webrtc/src/tokio/transport.rs:49:1
   |
49 |   pub struct Transport {
   |   --------------------
   |   |
   |   doesn't satisfy `libp2p_webrtc::tokio::Transport: Iterator`
   |   doesn't satisfy `libp2p_webrtc::tokio::Transport: StreamExt`
   |   doesn't satisfy `libp2p_webrtc::tokio::Transport: futures::Stream`
   |
   = note: the following trait bounds were not satisfied:
           `libp2p_webrtc::tokio::Transport: futures::Stream`
           which is required by `libp2p_webrtc::tokio::Transport: StreamExt`
           `&libp2p_webrtc::tokio::Transport: futures::Stream`
           which is required by `&libp2p_webrtc::tokio::Transport: StreamExt`
           `&mut libp2p_webrtc::tokio::Transport: futures::Stream`
           which is required by `&mut libp2p_webrtc::tokio::Transport: StreamExt`
           `libp2p_webrtc::tokio::Transport: Iterator`
           which is required by `&mut libp2p_webrtc::tokio::Transport: Iterator`
   = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
   |
3  + use libp2p_core::transport::Transport;
   |

warning: unused import: `futures::StreamExt`
 --> examples/browser-webrtc/src/main.rs:9:5
  |
9 | use futures::StreamExt;
  |     ^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: unused import: `core::Transport`
  --> examples/browser-webrtc/src/main.rs:12:5
   |
12 |     core::Transport,
   |     ^^^^^^^^^^^^^^^
```


### Possible Solution

I'm a bit stumped, as `libp2p::core` exports `Transport`, so I'm not really sure what's going on or why the traits would have issues based on the source of the library. The compile error even goes so far as to say that `core::Transport` is unused!

### Version

0.53.1

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -106,9 +106,9 @@ libp2p-tcp = { version = "0.41.0", path = "transports/tcp" }
 libp2p-tls = { version = "0.3.0", path = "transports/tls" }
 libp2p-uds = { version = "0.40.0", path = "transports/uds" }
 libp2p-upnp = { version = "0.2.0", path = "protocols/upnp" }
-libp2p-webrtc = { version = "0.6.1-alpha", path = "transports/webrtc" }
+libp2p-webrtc = { version = "0.7.1-alpha", path = "transports/webrtc" }
 libp2p-webrtc-utils = { version = "0.1.0", path = "misc/webrtc-utils" }
-libp2p-webrtc-websys = { version = "0.2.0-alpha", path = "transports/webrtc-websys" }
+libp2p-webrtc-websys = { version = "0.3.0-alpha", path = "transports/webrtc-websys" }
 libp2p-websocket = { version = "0.43.0", path = "transports/websocket" }
 libp2p-websocket-websys = { version = "0.3.1", path = "transports/websocket-websys" }
 libp2p-webtransport-websys = { version = "0.2.0", path = "transports/webtransport-websys" }

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -106,7 +106,7 @@ libp2p-tcp = { version = "0.41.0", path = "transports/tcp" }
 libp2p-tls = { version = "0.3.0", path = "transports/tls" }
 libp2p-uds = { version = "0.40.0", path = "transports/uds" }
 libp2p-upnp = { version = "0.2.0", path = "protocols/upnp" }
-libp2p-webrtc = { version = "0.7.1-alpha", path = "transports/webrtc" }
+libp2p-webrtc = { version = "0.7.0-alpha", path = "transports/webrtc" }
 libp2p-webrtc-utils = { version = "0.1.0", path = "misc/webrtc-utils" }
 libp2p-webrtc-websys = { version = "0.3.0-alpha", path = "transports/webrtc-websys" }
 libp2p-websocket = { version = "0.43.0", path = "transports/websocket" }

--- a/transports/webrtc-websys/CHANGELOG.md
+++ b/transports/webrtc-websys/CHANGELOG.md
@@ -1,7 +1,7 @@
 ## 0.3.0-alpha
 
 - Bump version in order to publish a new version dependent on latest `libp2p-core`.
-  See [PR 4959](https://github.com/libp2p/rust-libp2p/pull/4959)
+  See [PR 4959](https://github.com/libp2p/rust-libp2p/pull/4959).
 
 ## 0.2.0-alpha
 

--- a/transports/webrtc-websys/CHANGELOG.md
+++ b/transports/webrtc-websys/CHANGELOG.md
@@ -1,3 +1,8 @@
+## 0.3.0-alpha
+
+- Bump version in order to publish a new version dependent on latest `libp2p-core`.
+  See [PR 4959](https://github.com/libp2p/rust-libp2p/pull/4959)
+
 ## 0.2.0-alpha
 
 - Rename `Error::JsError` to `Error::Js`.

--- a/transports/webrtc-websys/Cargo.toml
+++ b/transports/webrtc-websys/Cargo.toml
@@ -8,7 +8,7 @@ license = "MIT"
 name = "libp2p-webrtc-websys"
 repository = "https://github.com/libp2p/rust-libp2p"
 rust-version = { workspace = true }
-version = "0.2.0-alpha"
+version = "0.3.0-alpha"
 publish = true
 
 [dependencies]

--- a/transports/webrtc/CHANGELOG.md
+++ b/transports/webrtc/CHANGELOG.md
@@ -1,3 +1,8 @@
+## 0.7.1-alpha
+
+- Bump version in order to publish a new version dependent on latest `libp2p-core`.
+  See [PR 4959](https://github.com/libp2p/rust-libp2p/pull/4959)
+
 ## 0.6.1-alpha
 
 - Move common dependencies to `libp2p-webrtc-utils` crate.

--- a/transports/webrtc/CHANGELOG.md
+++ b/transports/webrtc/CHANGELOG.md
@@ -1,4 +1,4 @@
-## 0.7.1-alpha
+## 0.7.0-alpha
 
 - Bump version in order to publish a new version dependent on latest `libp2p-core`.
   See [PR 4959](https://github.com/libp2p/rust-libp2p/pull/4959)

--- a/transports/webrtc/CHANGELOG.md
+++ b/transports/webrtc/CHANGELOG.md
@@ -1,7 +1,7 @@
 ## 0.7.0-alpha
 
 - Bump version in order to publish a new version dependent on latest `libp2p-core`.
-  See [PR 4959](https://github.com/libp2p/rust-libp2p/pull/4959)
+  See [PR 4959](https://github.com/libp2p/rust-libp2p/pull/4959).
 
 ## 0.6.1-alpha
 

--- a/transports/webrtc/Cargo.toml
+++ b/transports/webrtc/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "libp2p-webrtc"
-version = "0.7.1-alpha"
+version = "0.7.0-alpha"
 authors = ["Parity Technologies <admin@parity.io>"]
 description = "WebRTC transport for libp2p"
 repository = "https://github.com/libp2p/rust-libp2p"

--- a/transports/webrtc/Cargo.toml
+++ b/transports/webrtc/Cargo.toml
@@ -1,6 +1,6 @@
 [package]
 name = "libp2p-webrtc"
-version = "0.6.1-alpha"
+version = "0.7.1-alpha"
 authors = ["Parity Technologies <admin@parity.io>"]
 description = "WebRTC transport for libp2p"
 repository = "https://github.com/libp2p/rust-libp2p"

